presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ExpressionRewriteRuleSet.java

```diff
@@ -130,6 +130,33 @@ public Pattern getPattern()
             }
             return Optional.empty();
         }
+
+        private Optional<Aggregation> rewrite(Aggregation aggregation, Context context)
+        {
+            boolean anyRewritten = false;
+            ImmutableList.Builder<Expression> arguments = ImmutableList.builder();
+            for (Expression argument : aggregation.getCall().getArguments()) {
+                Expression rewritten = rewriter.rewrite(argument, context);
+                if (!argument.equals(rewritten)) {
+                    anyRewritten = true;
+                }
+                arguments.add(rewritten);
+            }
+            if (anyRewritten) {
+                return Optional.of(
+                        new Aggregation(
+                                new FunctionCall(
+                                        aggregation.getCall().getName(),
+                                        aggregation.getCall().getWindow(),
+                                        aggregation.getCall().getFilter(),
+                                        aggregation.getCall().isDistinct(),
+                                        arguments.build()
+                                ),
+                                aggregation.getSignature(),
+                                aggregation.getMask()));
+            }
+            return Optional.empty();
+        }
```

Comment:
This looks fine to me.  I think it could be shorter if the anyRewritten check was done by `equals` on the new `FunctionCall` object -- that way the new arguments could be constructed using streams.  But this is clearly not a vital change.